runelite-client/src/main/java/net/runelite/client/plugins/kingdomofmiscellania/KingdomPlugin.java

```diff
@@ -122,20 +189,60 @@ private void removeKingdomInfobox()
 		}
 	}
 
+	private int estimateCoffer(long lastChangedEpoch, int lastCoffer)
+	{
+		int daysSince = Math.max((int) (Instant.now().getEpochSecond() - lastChangedEpoch) / 86_400, 0);
+		int maxDailyWithdrawl = royalTroubleCompleted() ? 75_000 : 50_000;
+		int maxDailyThreshold = maxDailyWithdrawl * 10;
+
+		for (int i = 0; i < daysSince; i++)
+		{
+			lastCoffer -= (lastCoffer > maxDailyThreshold) ? maxDailyWithdrawl : lastCoffer / 10;
+		}
+		return lastCoffer;
+	}
+
+	private int estimateFavor(long lastChangedEpoch, int lastFavor)
+	{
+		int daysSince = Math.max((int) (Instant.now().getEpochSecond() - lastChangedEpoch) / 86_400, 0);
+		float dailyPercentage = royalTroubleCompleted() ? 0.01f : 0.025f;
+
+		lastFavor -= (int) (daysSince * dailyPercentage * 127);
```

Comment:
`127` should also be made into a constant and should get a healthy dose of javadoc to explain that approval rating is stored as an unsigned 7-bit int. (Perhaps this should be added to the varbit entry for kingdom favor)